src/core/writers/schemas.ts

```diff
@@ -85,18 +74,23 @@ export const writeSchemas = async ({
   specsName: Record<string, string>;
   header: string;
 }) => {
-  await ensureFile(join(schemaPath, '/index.ts'));
+  const schemaFilePath = join(schemaPath, '/index.ts');
+  await ensureFile(schemaFilePath);
 
-  return Promise.all(
-    schemas.map((schema) =>
-      writeSchema({
-        path: schemaPath,
-        schema,
-        target,
-        isRootKey,
-        specsName,
-        header,
-      }),
-    ),
+  schemas.map((schema) =>
+    writeSchema({
+      path: schemaPath,
+      schema,
+      target,
+      isRootKey,
+      specsName,
+      header,
+    }),
   );
+
+  try {
+    await writeFile(schemaFilePath, generateModelImports({ schemas }));
```

Comment:
The `appendFile` was there to not override the file completely. For example, if you already have some export in that file that you don't want orval to override each time. For example, having other models, not in the specification.

Reply:
Ah I understand now.

Is that a requirement or something you are willing to change?
If users want to add custom code they should probably be re-exporting in another barrel file.

Reply:
I am not sure anyone use that feature but I still prefer to maintain it because otherwise it's a breaking change

Reply:
Because this is a undocumented feature I normally use a minor release digit bump. But I understand if you want to use strict semantic versioning.

I can change the PR target to your next version but I don't see a `v7` branch and `next` is actually behind `master`. How would you like to proceed?

Reply:
> "Be carefull clean all output target and schemas folder."
----- https://orval.dev/reference/configuration/output#clean

The documentation on `clean` sort of mentions this feature... But I wasn't sure what that meant when initially integrating orval 😄

Reply:
We can merge it like that it is something that nobody knows and uses anyway